postmortems/2024-03-22_azure_dev_outage.md

```diff
@@ -0,0 +1,110 @@
+
+In addition, clients' Azure DevOps security permissions appear to have changed more broadly sometime after 6 September 2023 (the last traceable date where we can be absolutely certain MP & AL had access to the pieplines and repos on Azure Devops) without us being notified. MP & AL (again on B2B accounts) were either downgraded from being Basic users in devops down to merely Stakeholders, who cannot access the pipelines and repos, or the permissions for what Stakeholders can access was changed. Where pipelines and repos were previously visible to them in Azure Devops, they became not visible sometime after 6 September 2023. While not having directly contributed to the problem, this kind of practice could represent a risk. Subsequently, both senior developers’ B2B accounts have been upgraded to Basic Users and can now see and access Pipelines and Repos in DevOps. 
+
+MP has both a B2B and non-B2B account – and the permissions for each are different. The Dev environment has all the necessary permissions for the non-B2B (client) account, as this environment was setup second. The primary Prod environment was setup before Martin had a non-B2B (client) account and was therefore setup with the B2B account.  
+
+Subsequently, both of MP’s accounts (client and B2B) have been made Owners of the Prod app registration, as well as both of PO’s accounts. MP has added PO's NH account as an owner to the Dev app registration. We have also located all of the other app registrations and service principals and requested both MP and PO be added as owners to all of them.  
+
+### Why wasn’t this noticed on the Azure Dashboard? 
+Logging in to the client account, while also having a standard MS account for daily work as an employee has made logging into the dashboard as a regular habit less likely given the additional login required. Because there is friction to see what should be frictionless, it isn’t part of routine.  
+
+### Once the secret was obtained, why was access not immediately restored? 
+Line manager had indicated that change request governance processes would need to be followed to instate the new secret and restore access. Access was restored soon after the next CAB meeting occurred as permission was obtained during that meeting.  
+
+## Wonderings
+
+Why are B2B guest accounts added as app registration owners if they cannot exercise a primary privilege of app registration ownership? 
+
+Why did senior developers lose visibility of the pipelines and repos, and is this something that could happen again? Could similar loss of privilege occur without warning on other key areas? 
+
+Monitoring to detect the outage was ignored, it was only spotted when trying to access the service during working hours.
+
+There was clear and regular communication internally about the outage and progress to address it, and although initially stilted, eventually clear and regular communication with line manager (client side) on the matter. 
```

Comment:
I'm generally not a fan of closing with this section. I want to see the lessons learned and remediation actions at the end of a post-mortem, not open-ended questions.

Reply:
Added some remediations